Add Renesas GR-PEACH and GR-LYCHEE support

toyowata opened the pull request:
[GR-PEACH]

External SPI flash (memory mapped) : 8MB
RAM : 10MB
Connectivity : Ethernet

kvstore : memory mapped flash (48 kB)
Update storage : memory mapped flash (4MB)

[GR-LYCHEE]

External SPI flash (memory mapped) : 8MB
RAM : 3MB
Connectivity : Wi-Fi (ESP32)

kvstore : memory mapped flash (48 kB)
Update storage : memory mapped flash (4MB)

mbed-bootloader configs : https://github.com/ARMmbed/mbed-bootloader/pull/56


## Test result

### GR_PEACH (RZ_A1H)
```
============================== 4 passed in 31.78s ==============================
2020-01-15 11:05:10,999:tests.conftest:MainThread:INFO: -----  TEST RESULTS SUMMARY  -----
2020-01-15 11:05:11,000:tests.conftest:MainThread:INFO: [ check the complete fail reasons and code locations from this log or html report ]
2020-01-15 11:05:11,000:tests.conftest:MainThread:INFO: [passed] - dev-client-tests.py::test_01_get_device_id - (0.861s)
2020-01-15 11:05:11,000:tests.conftest:MainThread:INFO: [passed] - dev-client-tests.py::test_02_put_resource - (4.739s)
2020-01-15 11:05:11,000:tests.conftest:MainThread:INFO: [passed] - dev-client-tests.py::test_03_get_resource - (1.710s)
2020-01-15 11:05:11,001:tests.conftest:MainThread:INFO: [passed] - dev-client-tests.py::test_04_subscribe_resource - (2.451s)
```
### GR_LYCHEE
```
============================== 4 passed in 39.15s ==============================
2020-01-15 10:58:02,240:tests.conftest:MainThread:INFO: -----  TEST RESULTS SUMMARY  -----
2020-01-15 10:58:02,240:tests.conftest:MainThread:INFO: [ check the complete fail reasons and code locations from this log or html report ]
2020-01-15 10:58:02,241:tests.conftest:MainThread:INFO: [passed] - dev-client-tests.py::test_01_get_device_id - (0.772s)
2020-01-15 10:58:02,241:tests.conftest:MainThread:INFO: [passed] - dev-client-tests.py::test_02_put_resource - (4.831s)
2020-01-15 10:58:02,242:tests.conftest:MainThread:INFO: [passed] - dev-client-tests.py::test_03_get_resource - (1.839s)
2020-01-15 10:58:02,242:tests.conftest:MainThread:INFO: [passed] - dev-client-tests.py::test_04_subscribe_resource - (3.470s)
```

iot-pdmc commented:
Thanks for your changes.
@ARMmbed/pdmc-integrators please review.

teetak01 commented on drivers/COMPONENT_WIFI_ESP32.lib:
```diff
@@ -0,0 +1 @@
+https://github.com/ARMmbed/esp32-driver/#d6bd83f8710b4c475b3cce101d8b2c77e17feaf6
```
Could you use here the COMPONENT format as has been done for the existing drivers?

https://github.com/ARMmbed/mbed-os-example-pelion/tree/master/drivers

toyowata replied:
I am not clear about your comment, but the GR-LYCHEE has on-board ESP32 Wi-Fi module and it uses existing esp32-driver.

teetak01 replied:
Ah, sorry, I misread, you already did it as expected.

teetak01 commented on mbed_app.json:
```diff
@@ -244,6 +244,44 @@
+            "update-client.storage-size"                : "(4*1024*1024-MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_SIZE)",
+            "update-client.storage-page"                : 1,
+            "mbed-cloud-client.update-storage"          : "ARM_UCP_FLASHIAP",
+            "storage_tdb_internal.internal_base_address": "(MBED_ROM_START+4*1024*1024)",
```
Does this board really have 8 MB of internal flash? Is it working in dual-bank mode? This configuration would suggest that it is. Mbed OS documentation seems to say so, but Renessas own documentation does not make very clear mention at all.

JanneKiiskila replied:
External flash is not as secure as internal flash, so probably some warning caveats should be put in place. Do the chips have any OTP key functionality?

JanneKiiskila replied:
README.md needs to be thus updated with some proper boiler plate (and added to the list of supported devices).

toyowata replied:
I agreed to update the README.md to add some warning about the external flash.

teetak01 commented:
Does this depend on https://github.com/ARMmbed/mbed-os/pull/12223 ?

toyowata commented:
@teetak01 

> Does this depend on ARMmbed/mbed-os#12223 ?

I put the workaround here in this patch : https://github.com/toyowata/mbed-os-example-pelion-1/blob/215f506e605c677b11cdf06f0345f3c0ba3136fd/mbed_app.json#L251

This can be removed from the mbed_app.json once the https://github.com/ARMmbed/mbed-os/pull/12223 will be merged and the mbed-os-example-pelion will use it.

yogpan01 commented:
@toyowata Please rebase and resolve merge conflict from this PR.

toyowata commented:
@yogpan01 @teetak01 

* Rebased and resolve merge conflict
* Added test results https://github.com/ARMmbed/mbed-os-example-pelion/pull/83#issue-360279665
* Added targets in the README.md with notification about the external flash usage

teetak01 commented:
build-only

iot-pdmc commented:
# [Test run: SUCCESS](https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/570/)
***
[Build artifacts](http://pdmc-ci.s3-website-eu-west-1.amazonaws.com/?prefix=jenkins-ci/mbed-os-example-pelion/570)

posted-by-tag: https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/

JanneKiiskila commented on README.md:
```diff
@@ -27,6 +27,8 @@ Board                          |  Connectivity     | Storage for credentials and
 `ST NUCLEO_L4R5ZI`                  | Wi-Fi ESP8266     | Internal Flash          | Build-only
 `Nuvoton NUMAKER_IOT_M487`          | Wi-Fi ESP8266     | SD card (NUSD)          | Build-only
+`Renesas RZ_A1H`                    | Ethernet          | External Flash (Note that it is not as secure as internal Flash)         | Build-only
+`Renesas GR_LYCHEE`                    | Wi-Fi (ESP32)  | External Flash (Note that it is not as secure as internal Flash)         | Build-only
```
I would want the more explicit statements about production devices using the specially ordered secure versions of the chips being available from Renesas. The board info pages do not mention that;

- https://os.mbed.com/platforms/Renesas-GR-LYCHEE/
- https://os.mbed.com/platforms/Renesas-GR-PEACH/

Might make most sense to have it directly on those pages and just link to them?

toyowata replied:
@JanneKiiskila I added security limitation info for both board page and modified readme.

toyowata commented:
@teetak01 rebased.

toyowata commented:
Created new https://github.com/ARMmbed/mbed-os-example-pelion/pull/96 as this branch broken.

iot-pdmc commented:
# [Test run: SUCCESS](https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/575/)
***
[Build artifacts](http://pdmc-ci.s3-website-eu-west-1.amazonaws.com/?prefix=jenkins-ci/mbed-os-example-pelion/575)

posted-by-tag: https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/
